openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -0,0 +1,196 @@
+"""
+API for managing user preferences.
+"""
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.db import transaction
+from django.utils.translation import ugettext as _
+
+from ..api.user import UserApiInternalError, UserApiRequestError, UserNotFound, UserNotAuthorized
+from ..helpers import intercept_errors
+from ..models import UserPreference, PreferenceNotFound, PreferenceValidationError, PreferenceUpdateError
+from ..serializers import UserSerializer
+
+
+def _get_user(requesting_user, username=None, allow_staff=False):
+    """
+    Helper method to return the user for a given username.
+    If username is not provided, requesting_user.username is assumed.
+    """
+    if username is None:
+        username = requesting_user.username
+
+    try:
+        existing_user = User.objects.get(username=username)
+    except ObjectDoesNotExist:
+        raise UserNotFound()
+
+    if requesting_user.username != username:
+        if not requesting_user.is_staff or not allow_staff:
+            raise UserNotAuthorized()
+
+    return existing_user
+
+
+@intercept_errors(UserApiInternalError, ignore_errors=[UserApiRequestError])
```

Comment:
I would expect the ignore error to be PreferenceRequestError. I think I need to draw a diagram of the various errors so we know which ones should be returned at each place.